workflow/controller/operator.go

```diff
@@ -1008,9 +1008,18 @@ func (woc *wfOperationCtx) processNodeRetries(node *wfv1.NodeStatus, retryStrate
 		node = woc.markNodePhase(node.Name, node.Phase, "")
 	}
 
+	retryPolicy := retryStrategy.RetryPolicy
+	if retryPolicy == "" {
+		if retryStrategy.Expression == "" {
+			retryPolicy = wfv1.RetryPolicyOnFailure
+		} else {
+			retryPolicy = wfv1.RetryPolicyAlways
+		}
+	}
+
```

Comment:
maybe we can add a log line here so it's obvious the result?

Reply:
Ok, done.